src/app/services/date/date.service.ts

```diff
@@ -1,10 +1,42 @@
-import { Injectable } from '@angular/core';
+import { formatNumber } from '@angular/common';
+import { inject, Injectable, LOCALE_ID } from '@angular/core';
+import { Column } from 'angular-google-charts';
 import { addDays, differenceInDays, format, isEqual } from 'date-fns';
+import { ChartData, RegistryData } from '../../models';
 
 @Injectable({
 	providedIn: 'root',
 })
 export class DateService {
+	private readonly locale = inject(LOCALE_ID);
+
+	getChartData(apiDatas: RegistryData[], start: Date, end: Date): ChartData {
+		const columns: Column[] = [
+			{ type: 'string', label: 'Date' },
+			...apiDatas.map(({ packageName, total }) => ({
+				type: 'number',
+				label: `${packageName} (${formatNumber(total, this.locale)})`,
+			})),
+		];
+
+		const dates = this.getDateRange(start, end);
+
+		// TODO: get formatter given range
+		// MM/dd vs MM/dd/yy
+		const rows = dates.map((date, i) => {
```

Comment:
This logic should be the only part that's migrated to `DateService`. Currently it works off of an assumption that `getDownloadsRange` returns downloads in the right order for this section.

Can we also refactor `getDownloadsRange` to return `{downloads: number, day: string}[]`, then update this logic find an apiData by comparing to day property (fallback to 0)?

I believe we should do this for 2 reasons:
1. We can later improve caching system
2. We have no control on the `getDownloadsRange` api response

Let me know what you think of this, after we sync I'll update the jira ticket for the change of scope

Reply:
I agree of reworking the `getDownloadsRange()` to return `[{downloads: number, day: string}]` instead of just a list of numbers. I believe it will be easier to calculate weekly or monthly increases

Reply:
I added this change in [d2aa9d9](https://github.com/bitovi/opensource-analytics-dashboard/pull/11/commits/d2aa9d93d4e1ea673c2803c6ab2e3a5a9eefcaaf)